test/resources/selinux.py

```diff
@@ -112,6 +150,127 @@ def run_default_access_check() -> List[str]:
+        return error_list
+    return run_binary_domain_transition_check(script_file_path, ACCESS_CHECK_MAP)
+
+
+# For a given root source, query transition process types to recursivly get all
+# contexts
+def get_all_traversal_transition_contexts(source: str) -> List[str]:
+    sesearch_cmd = "sesearch --type_trans -c process -s"
+    accum = []
+    # Search for a process transition from the specified source
+    stdout, rc = remote_sudo_rc(f"{sesearch_cmd} {source} | awk '{{ print $3 }}' 2>&1")
+    BuiltIn().should_be_equal_as_integers(rc, 0)
+
+    transition_contexts = stdout.splitlines()
+    if len(transition_contexts) == 0:
+        return accum
+
+    for context in transition_contexts:
+        source = context.split(":")[0]
+        accum.append(source)
+        result = get_all_traversal_transition_contexts(source)
+        accum.extend(result)
+
+    # Remove dups
+    return list(set(accum))
+
+
+# Check if we should ignore this specific domain check
+def ignore_domain_check(source: str, key: str, domain_permissions: List[str]) -> bool:
+    # Ignore self refernce to filesystem associatation
+    found = re.search(f"{source} {source}:filesystem", key)
+    if found:
+        if len(domain_permissions) == 1 and "associate" in domain_permissions:
+            print(f"match on {source} key {key}")
+            return True
+
+    # Iterate over ignore rules
+    for reg, permissions in DOMAIN_PERMISSION_IGNORE_REGEX.items():
+        found = re.search(reg, key)
+        if found and all(perm in permissions for perm in domain_permissions):
+            print(f"match on {reg} key {key}")
+            return True
+    return False
+
+
+# Given a parent context and all of its possible transitional context sources check to make sure
+# each one of the transitional sources does not contain permissions that are not expected or more
+# than the parent permissions.
+# i.e. a permission to a context that the parent does not have.
+def sources_have_less_permission_than_parent(parent: str, all_sources: List[str]) -> List[str]:
+    allow_cmd = "sesearch -S -A"
+    semanage_cmd = "semanage fcontext -l"
+
+    parent_number_of_permissions, rc = remote_sudo_rc(f"{allow_cmd} -s {parent} | wc -l")
+    BuiltIn().should_be_equal_as_integers(rc, 0)
+    error_list = []
+
+    for source in all_sources:
+        domain_context = source
+        if "_exec_t" in domain_context:
+            domain_context = source.replace('_exec', '')
+
+        source_number_of_permissions, rc = remote_sudo_rc(f"{allow_cmd} -s {domain_context} | wc -l")
+        BuiltIn().should_be_equal_as_integers(rc, 0)
+
+        if source_number_of_permissions != parent_number_of_permissions:
+            file_path, rc = remote_sudo_rc(f"{semanage_cmd} | grep {source} | awk '{{ print $1 }}'")
+            BuiltIn().should_be_equal_as_integers(rc, 0)
+            file_exists, rc = remote_sudo_rc(f"[ -e {file_path} ] && echo found || echo safe")
```

Comment:
I think this part of the logic will fail if `semanage fcontext -l` returns more than one path that's labeled with {source}.

Reply:
Ah, good catch, let me fix that real quick

Reply:
Updated